rabbitmq/src/main/java/io/micronaut/rabbitmq/intercept/RabbitMQConsumerAdvice.java

```diff
@@ -372,11 +293,233 @@ private void handleException(RabbitListenerException exception) {
+                                consumerTag);
+                        cancel();
+                    }
+                }
+
+                /**
+                 * A shutdown signal from the channel or the underlying connection does not always imply that the
+                 * consumer is no longer usable. If the automatic topology recovery is active and the shutdown
+                 * was not initiated by the application it will be recovered by the RabbitMQ client.
+                 * <p>
+                 * If the topology recovery is enabled we will also try to recover the consumer if (only) its channel
+                 * fails. These events are not handled by the RabbitMQ client itself as they are most likely application
+                 * specific. Also some edge cases like a delivery acknowledgement timeout may cause a channel shutdown.
+                 * The registered exception handler of the consumer may handle these cases and it is possible to
+                 * resume message handling by re-registering the consumer on a new channel.
+                 */
+                @Override
+                public void handleShutdownSignal(String consumerTag, ShutdownSignalException sig) {
+                    if (isDown) {
+                        return;
+                    }
+                    if (getChannel() instanceof RecoverableChannel && sig.isHardError()) {
+                        // The RabbitMQ client automatic recovery is only triggered by connection errors.
+                        // The consumer will be recovered by the client, so no additional handling here.
+                        LOG.info("The underlying connection was terminated. Automatic recovery attempt is underway for consumer [{}]",
+                                consumerTag);
+                    } else if (channelPool.isTopologyRecoveryEnabled() && getChannel() instanceof RecoverableChannel) {
+                        isDown = true;
+                        synchronized (RecoverableConsumerWrapper.this) {
+                            RecoverableConsumerWrapper.this.consumer = null;
+                            channelPool.returnChannel(getChannel());
+                        }
+                        performConsumerRecovery();
+                    } else {
+                        LOG.error("The channel was closed. Recovery is not enabled. The consumer [{}] will no longer receive messages",
+                                consumerTag, sig);
+                        cancel();
+                    }
+                }
+
+                @Override
+                public void handleDelivery(String consumerTag, Envelope envelope, AMQP.BasicProperties properties, byte[] body)
+                        throws IOException {
+                    if (!getChannel().isOpen()) {
+                        return;
+                    }
```

Comment:
Why do we care if the channel is open at this point? For acknowledgements? Shouldn't this be logged or an exception be thrown here instead of silently ignoring the issue?

Reply:
See my added implementation comment for an explanation.

I see no advantage in writing exceptions to the log when i can foresee what happens. And as i wrote in the comment, the message is not lost.

Reply:
I just re-tested the timeout without the prevention. And it seems that you are right.
The recovery works. This is a bit of over optimizing.

Reply:
Makes sense thanks